osu.Game/Overlays/Dashboard/CurrentlyPlayingDisplay.cs

```diff
@@ -54,15 +58,20 @@ protected override void LoadComplete()
                 switch (e.Action)
                 {
                     case NotifyCollectionChangedAction.Add:
-                        foreach (var u in e.NewItems.OfType<int>())
+
+                        foreach (var id in e.NewItems.OfType<int>().ToArray())
                         {
-                            var request = new GetUserRequest(u);
-                            request.Success += user => Schedule(() =>
+                            users.GetUser(id).ContinueWith(u =>
                             {
-                                if (playingUsers.Contains(user.Id))
-                                    userFlow.Add(createUserPanel(user));
+                                if (u.Result == null)
+                                    return;
+
+                                Schedule(() =>
+                                {
+                                    if (playingUsers.Contains(u.Result.Id))
+                                        userFlow.Add(createUserPanel(u.Result));
+                                });
```

Comment:
I'd probably split a method somewhere here - the nesting is nearing on too much I'd say.